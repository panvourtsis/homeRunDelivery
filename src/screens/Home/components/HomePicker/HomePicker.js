import React from 'react'
import PropTypes from 'prop-types'
import { Text, View } from 'react-native'
import ModalDropdown from 'react-native-modal-dropdown'
import Feather from 'react-native-vector-icons/Feather'
import styles from './HomePicker.style'

const data = ['tesco', 'waitrose']

const HomePicker = ({selectedSupplier, supplierPickerChange}) => (
  <View style={styles.container}>
    <ModalDropdown
      options={data}
      onSelect={(index, value) => supplierPickerChange(value)}
      defaultIndex={1}
      defaultValue={data[0]}
      dropdownStyle={styles.dropDown}
    >
      <View style={styles.dropDownContainer}>
        <Text style={styles.dropDownText}>{selectedSupplier}</Text>
        <Feather
          name='chevron-down' size={25} color='#A1A1A1'
        />
      </View>
    </ModalDropdown>
  </View>
)

HomePicker.propTypes = {
  selectedSupplier: PropTypes.string,
  supplierPickerChange: PropTypes.func
}

export default HomePicker
