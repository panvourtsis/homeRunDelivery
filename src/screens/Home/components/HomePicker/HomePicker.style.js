const styles = {
  container: { flex: 1, flexDirection: 'row', alignItems: 'center', justifyContent: 'center', alignSelf: 'center' },
  pickerStyle: { marginRight: 10, alignSelf: 'center', backgroundColor: 'transparent' },
  dropDown: {
    padding: 10,
    width: 100,
    height: 90
  },
  dropDownContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center'
  },
  dropDownText: { color: '#A1A1A1' }
}

export default styles
