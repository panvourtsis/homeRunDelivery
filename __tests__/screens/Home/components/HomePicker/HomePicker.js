import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import HomePicker from '@screens/Home/components/HomePicker/HomePicker'

describe('Testing HomePicker', () => {
  let wrapper
  const spyFn = jest.fn()
  beforeEach(() => {
    wrapper = shallow(
      <HomePicker selectedSupplier={'tesco'} supplierPickerChange={spyFn} />
    )
  })
  it('renders as expected', () => {
    expect(wrapper.dive()).toMatchSnapshot()
  })
  it('changes the value on change', () => {
    wrapper.dive().find('ModalDropdown').simulate('select')
    expect(spyFn).toHaveBeenCalledTimes(1)
  })
})
