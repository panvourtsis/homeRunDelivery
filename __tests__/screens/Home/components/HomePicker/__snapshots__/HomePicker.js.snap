// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing HomePicker renders as expected 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <ModalDropdown
    animated={true}
    defaultIndex={1}
    defaultValue="tesco"
    disabled={false}
    dropdownStyle={
      Object {
        "height": 90,
        "padding": 10,
        "width": 100,
      }
    }
    keyboardShouldPersistTaps="never"
    onSelect={[Function]}
    options={
      Array [
        "tesco",
        "waitrose",
      ]
    }
    scrollEnabled={true}
    showsVerticalScrollIndicator={true}
  >
    <Component
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#A1A1A1",
          }
        }
      >
        tesco
      </Text>
      <Icon
        allowFontScaling={false}
        color="#A1A1A1"
        name="chevron-down"
        size={25}
      />
    </Component>
  </ModalDropdown>
</View>
`;
